okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.kt

```diff
@@ -200,6 +200,7 @@ class AndroidPlatform : Platform() {
 
     val isSupported: Boolean = when {
       !isAndroid -> false
+      Build.VERSION.SDK_INT >= 30 -> false // graylisted methods are banned
```

Comment:
I think the failure is not related to using AndroidPlatform instead of Android10Platform, but rather still possibly building AndroidPlatform on versions >= R

Reply:
This shouldn't be needed but is correct IIUC

Reply:
Could we replace this with a check?

```
check(Build.VERSION.SDK_INT < 30)
```

Reply:
A check seems wrong for an isSupported call.  So will avoid this.